pkg/kubelet/lifecycle/handlers.go

```diff
@@ -47,6 +49,11 @@ const (
 
 	AppArmorNotAdmittedReason          = "AppArmor"
 	PodLevelResourcesNotAdmittedReason = "PodLevelResourcesNotSupported"
+
+	// Reasons for pod features admission failure
+	FeatureRequirementInferError = "FeatureRequirementInferError"
+	FeatureMatchError            = "FeatureMatchError"
+	MissingNodeDeclaredFeatures  = "MissingNodeDeclaredFeatures"
```

Comment:
nit: "NodeDeclaredFeatures" is the feature name for development purposes, but this is a user-visible message. I think something like `PodFeatureUnsupported` might be better.